src/box/lua/net_box.lua

```diff
@@ -1149,6 +1149,11 @@ end
 
 function remote_methods:ping(opts)
     check_remote_arg(self, 'ping')
+
+    if opts ~= nil then
+        box.error(box.error.UNSUPPORTED, "ping", "options")
```

Comment:
One may actually pass options to `ping`, e.g. you can specify a timeout: `conn:ping{timeout = 5}`. BTW there are tests that use this, and they would be broken by your change:

https://github.com/tarantool/tarantool/blob/06c0ebbd8023d750106e86807605b175320ec121/test/box/net.box_iproto_id.test.lua#L66

The problem described in #6063 is that one passes a number instead of a table for `opts` while `remote_methods:_request`, which is called by `ping` expects `opts` to be a table:

https://github.com/tarantool/tarantool/blob/06c0ebbd8023d750106e86807605b175320ec121/src/box/lua/net_box.lua#L1099-L1105

The caller doesn't get an exception, because `ping` calls `remote_methods:_request` with `pcall` and ignores the error. We could probably check the type of `opts` in `ping` to fix this.

Reply:
Thanks for review! I've added additional check if opts is a table as suggested.

Reply:
1. The commit is missing a [changelog entry](https://www.tarantool.io/en/doc/latest/contributing/release_notes/) and a test. For an example, see 2bbd13b45e500912cf2412e7ea618d126ecaf4f9.
2. Using `box.error.UNSUPPORTED` for invalid arguments isn't right. For an example, look at how we report invalid arguments [here](https://github.com/tarantool/tarantool/blob/330100bf2c369f0800100680966139806d13fea6/src/box/lua/net_box.lua#L594), [here](https://github.com/tarantool/tarantool/blob/330100bf2c369f0800100680966139806d13fea6/src/box/lua/net_box.lua#L429), and [here](https://github.com/tarantool/tarantool/blob/330100bf2c369f0800100680966139806d13fea6/src/box/lua/net_box.lua#L444). We're inconsistent: sometimes we use `ER_PROC_LUA` sometimes plain `error()`. You can use either of them, but definitely not `box.error.UNSUPPORTED`.
3. **Optional**: other net.box methods (e.g. `remote.call`) are affected to the issue just as well (although they don't silently swallow the error, as `ping` does because of `pcall`):
   ```
   tarantool> c:call('foo', {}, 123)
   ---
   - error: 'builtin/box/net_box.lua:694: attempt to index local ''opts'' (a number value)'
   ...
   ```
   It'd be nice to fix them all in the scope of this issue.